Switch to CentOS Stream 9

andreabolognani opened the pull request:
**What this PR does / why we need it**:

Now that CentOS Stream 9 has been out for a while, the pace of development for CentOS Stream 8 is slowing down and we generally shouldn't rely on major updates to the virtualization stack happening as regularly as they have in the past.

In order to ensure that KubeVirt can keep benefiting from improvements made to the various virtualization components going forward, move from CentOS Stream 8 to CentOS Stream 9.

**Special notes for your reviewer**:

The PR is marked as a draft and, before we can consider merging it, at least two things need to happen:

1. a new release of bazeldnf that contains https://github.com/rmohr/bazeldnf/pull/22 needs to be tagged;
2. someone needs to push an official CentOS Stream 9-based builder image.

I also expect CI to detect issues that I have not noticed during my own light testing and that it will take a few respins to address those.

Note that I have included the changes from https://github.com/kubevirt/kubevirt/pull/8403 in this PR because, while those changes are useful on their own, this PR depends on them.

**Release note**:

```release-note
This version of KubeVirt includes upgraded virtualization technology based on libvirt 8.7.0, QEMU 7.1.0 and CentOS Stream 9.
Each new release of libvirt, QEMU and CentOS Stream contains numerous improvements and bug fixes.
```

andreabolognani commented:
/test all

rmohr commented:
/retest

rmohr commented:
Seems like with the centos9 base we have again a `python` detection problem:

```
env: 'python': No such file or directory
```

A symlink may be able to fix it. Let me also see if we should just go to bazel 5, which I think has much less issues with python2/3.

alicefr commented:
a bit off-topic, I know, but do you plan also to update [kubevirtci](https://github.com/kubevirt/kubevirtci/tree/main/cluster-provision/centos8) from centos 8 to centos stream 9?

andreabolognani commented:
/test all

rmohr commented:
> which is also surprising - it seems to be still using the old, CentOS Stream 8 based toolchain instead of the CentOS Stream 9 one. I wonder if there's some cached data that needs to be nuked or something?

Oh I think that one is easy to fix. :) We need to update this line: https://github.com/kubevirt/kubevirt/commit/2481a0b05eaf7db3675b5d9edb1947466534eb1f#diff-0f7faf4e1647db546c2de02eca16f80fd190e9da6472bd99e88f561d9aa8d532R112

andreabolognani commented:
@rmohr I haven't been able to reproduce either failure locally.

Most of the failures are happening in `make bazel-build-images`, which is before we get to run any of the actual functional tests. I have made the name of the (temporary) CentOS Stream 9 based builder image more unique and added some debug prints to the scripts.

Local execution:

```
make bazel-build-images 
hack/dockerized "export BUILD_ARCH= && DOCKER_PREFIX= DOCKER_TAG= DOCKER_TAG_ALT= IMAGE_PREFIX= IMAGE_PREFIX_ALT= ./hack/bazel-build-images.sh"
Using quay.io/andreabolognani/kubevirt-builder:2209071113-9d667f192
go version go1.17.8 linux/amd64

go version go1.17.8 linux/amd64
Sandbox is up to date
NAME="CentOS Stream"
VERSION="9"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="9"
PLATFORM_ID="platform:el9"
PRETTY_NAME="CentOS Stream 9"
ANSI_COLOR="0;31"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:centos:centos:9"
HOME_URL="https://centos.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux 9"
REDHAT_SUPPORT_PRODUCT_VERSION="CentOS Stream"
python is /usr/bin/python
lrwxrwxrwx. 1 root root 16 Sep  7 09:14 /usr/bin/python -> /usr/bin/python3
INFO: Build option --define has changed, discarding analysis cache.
[...]
```

CI output:

```
 + make bazel-build-images
"Timestamp is enabled"
hack/dockerized "export BUILD_ARCH= && DOCKER_PREFIX= DOCKER_TAG= DOCKER_TAG_ALT= IMAGE_PREFIX= IMAGE_PREFIX_ALT=kv- ./hack/bazel-build-images.sh"
10:26:30: Regenerating sandbox
[...]
10:27:24: INFO: Running command line: bazel-bin/rpm/sandbox_x86_64.bash
10:27:24: INFO: Build completed successfully, 307 total actions
10:27:25: INFO: Invocation ID: 264dc48c-cf05-4eb1-9a15-7bb958bb0e5c
10:27:25: INFO: Starting clean (this may take a while). Consider using --async if the clean takes more than several minutes.
10:27:26: NAME="Fedora Linux"
10:27:26: VERSION="36 (Container Image)"
10:27:26: ID=fedora
10:27:26: VERSION_ID=36
10:27:26: VERSION_CODENAME=""
10:27:26: PLATFORM_ID="platform:f36"
10:27:26: PRETTY_NAME="Fedora Linux 36 (Container Image)"
10:27:26: ANSI_COLOR="0;38;2;60;110;180"
10:27:26: LOGO=fedora-logo-icon
10:27:26: CPE_NAME="cpe:/o:fedoraproject:fedora:36"
10:27:26: HOME_URL="https://fedoraproject.org/"
10:27:26: DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f36/system-administrators-guide/"
10:27:26: SUPPORT_URL="https://ask.fedoraproject.org/"
10:27:26: BUG_REPORT_URL="https://bugzilla.redhat.com/"
10:27:26: REDHAT_BUGZILLA_PRODUCT="Fedora"
10:27:26: REDHAT_BUGZILLA_PRODUCT_VERSION=36
10:27:26: REDHAT_SUPPORT_PRODUCT="Fedora"
10:27:26: REDHAT_SUPPORT_PRODUCT_VERSION=36
10:27:26: PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
10:27:26: VARIANT="Container Image"
10:27:26: VARIANT_ID=container
10:27:26: python is /usr/bin/python
10:27:26: lrwxrwxrwx. 1 root root 9 Jun  9 22:28 /usr/bin/python -> ./python3
10:27:26: INFO: Invocation ID: 22dd4db8-d3fc-45f7-8c6a-86b39cdcc55c
[...]
```

So the build is indeed not happening in the CentOS Stream 9 container, but in a Fedora 36 one. I believe this is a consequence of `KUBEVIRT_RUN_UNNESTED=true` being set in the CI job's environment, and this seems to be confirmed by the fact that we're not seeing the "Using quay.io/..." message in the second output: the `hack/dockerized` script ends early.

This still doesn't explain why the failure is about a missing Python binary, since the Fedora 36 image contains it, but at this point I think I need to take a step back. If the CI build happens inside a completely different container image than the one used for local builds, how can I reproduce and debug the issue locally? And is this difference intentional? It certainly was unexpected to me.

andreabolognani commented:
> Oh I think that one is easy to fix. :) We need to update this line: [2481a0b#diff-0f7faf4e1647db546c2de02eca16f80fd190e9da6472bd99e88f561d9aa8d532R112](https://github.com/kubevirt/kubevirt/commit/2481a0b05eaf7db3675b5d9edb1947466534eb1f#diff-0f7faf4e1647db546c2de02eca16f80fd190e9da6472bd99e88f561d9aa8d532R112)

@rmohr I've done exactly that [here](https://github.com/kubevirt/kubevirt/pull/8404/commits/7f4a36af0bad8d3a1db287e8f8c4356466af0e23#diff-0f7faf4e1647db546c2de02eca16f80fd190e9da6472bd99e88f561d9aa8d532L112-R112) ("bazel: Update toolchains for CentOS Stream 9").

The path goes from 8 to 11 because CentOS Stream 9 uses GCC 11 instead of GCC 8, which CentOS Stream 8 was using.

But note that bazel complains about the files for GCC 8 not being listed among requirements, which means that somehow the old toolchain is being used. I have no idea why that would be the case.

At least the expected builder image is used here, but as the toolchain sandbox is handled independently that doesn't help.

andreabolognani commented:
Rebased and refreshed all contents, including the temporary CentOS Stream 9-based builder image. bazeldnf 0.5.3 has been tagged and contains all the changes we need, so the corresponding commit is no longer marked as DONOTMERGE.

I still expect the same failures as before to show up in CI, but let's wait and see...

andreabolognani commented:
Okay, the latest test attempts failed, just as I was expecting.

I tried once again to reproduce the failure locally. Specifically, I took the [definition](https://github.com/kubevirt/project-infra/blob/6c743bab1dc7b5273de5a2b26845267d1c3a9e5b/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml#L904-L932) for the `pull-kubevirt-job`, which looks like this

```yaml
- always_run: true
  annotations:
    fork-per-release: "true"
  cluster: ibm-prow-jobs
  decorate: true
  decoration_config:
    grace_period: 5m0s
    timeout: 1h0m0s
  labels:
    preset-bazel-cache: "true"
    preset-podman-in-container-enabled: "true"
    preset-docker-mirror-proxy: "true"
  name: pull-kubevirt-build
  skip_branches:
  - release-\d+\.\d+
  spec:
    containers:
    - command:
      - /usr/local/bin/runner.sh
      - /bin/sh
      - -c
      - cp /etc/bazel.bazelrc ./ci.bazelrc && make && make build-verify
      image: quay.io/kubevirtci/bootstrap:v20220906-4f60dd3
      name: ""
      resources:
        requests:
          memory: 4Gi
      securityContext:
        privileged: true
```

and based on it I started a local container with

```
$ docker run --privileged --entrypoint bash --rm -it quay.io/kubevirtci/bootstrap:v20220906-4f60dd3
```

Inside the container, I then ran

```
$ cd ../
$ git clone https://github.com/andreabolognani/kubevirt.git
$ mv workspace/gcloud-info.txt kubevirt/ && rmdir workspace/ && mv kubevirt/ workspace/
$ cd workspace/
$ git checkout centos-stream-9
$ export PODMAN_IN_CONTAINER_ENABLED=true
$ export PODMAN_IN_CONTAINER_IPV6_ENABLED=false
$ /usr/local/bin/runner.sh /bin/sh -c 'cp /etc/bazel.bazelrc ./ci.bazelrc && make'
```

The output looked like

```
Enabling IPV6.
net.ipv6.conf.all.disable_ipv6 = 0
net.ipv6.conf.all.forwarding = 1
modprobe: FATAL: Module ip6table_nat not found in directory /lib/modules/5.19.12-200.fc36.x86_64
Podman in Container enabled, initializing in podman compatible mode...
Waiting for podman socket to be ready, sleeping for 1 seconds.
================================================================================
Done setting up podman in container.
+ /bin/sh -c 'cp /etc/bazel.bazelrc ./ci.bazelrc && make'
./hack/dockerized "hack/bazel-fmt.sh"
Using quay.io/andreabolognani/kubevirt-builder:2210061113-f928b73d3
kubevirt
Trying to pull quay.io/andreabolognani/kubevirt-builder:2210061113-f928b73d3...
...
```

which is similar to that of the most recent `pull-kubevirt-build` attempt - except in this case the build completes successfully.

Note that one important difference is that, for my local build, I'm not enabling bazel remote caching. My impression is that caching is exactly what's getting in the way here, because when you look at the [log](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/8404/pull-kubevirt-build/1578033048107094016) for the CI job you can see that the correct container is being used, and yet the compiler is searching the wrong (outdated) directories when looking for standard headers...

```
ERROR: /root/go/src/kubevirt.io/kubevirt/cmd/container-disk-v2alpha/BUILD.bazel:3:10: Compiling cmd/container-disk-v2alpha/main.c failed: undeclared inclusion(s) in rule '//cmd/container-disk-v2alpha:container-disk':
this rule is missing dependency declarations for the following files included by 'cmd/container-disk-v2alpha/main.c':
  '/usr/lib/gcc/x86_64-redhat-linux/8/include/stdarg.h'
  '/usr/lib/gcc/x86_64-redhat-linux/8/include/stddef.h'
  '/usr/lib/gcc/x86_64-redhat-linux/8/include/stdbool.h' 
```

Am I completely off? Can anyone with better understanding of the CI infrastructure (_\*cough\* @rmohr \*cough\*_) help shed some light?

xpivarc commented:
> ```yaml
> preset-bazel-cache
> ```

Let's try to override `BAZEL_REMOTE_CACHE_ENABLED` in the make target.

andreabolognani commented:
> Let's try to override `BAZEL_REMOTE_CACHE_ENABLED` in the make target.

@xpivarc very good idea, I'll give it a try!

rmohr commented:
If it is just a matter of the bazel-cache, we should be able to tweak that. We have a function which bazel uses to create a tool-hash-key which is used to determine a matching cache bucket. Would be great to try it out without cache. If it then passes I can check which parts exactly we need to add to have the right bit considered in the tool-hash-key. Thanks!!

andreabolognani commented:
/retest pull-kubevirt-build

andreabolognani commented:
/test pull-kubevirt-build

andreabolognani commented:
/test pull-kubevirt-e2e-k8s-1.22-sig-compute

andreabolognani commented:
Getting the bazel cache out of the picture did the trick: [successful `pull-kubevirt-build` job](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/8404/pull-kubevirt-build/1579405223602950144).

Of course that only proves that the problematic interaction is with the cache and is not a solution. How do I dig further? Can I build a local cache that is similar in terms of contents to the one used by CI jobs and trace exactly what is being requested from it in order to track down the failure? If so, how? Any other ideas? @xpivarc @rmohr

The Python issue is still present: [unsuccessful `pull-kubevirt-e2e-k8s-1.22-sig-compute` job](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/8404/pull-kubevirt-e2e-k8s-1.22-sig-compute/1579415199645962240).

I just noticed however that, in this case, the failure occurs while handling the libguestfs image, which uses a pre-built appliance that's based on CentOS Stream 8. I think generating an appliance that's based on CentOS Stream 9 instead might help. I'm going to be investigating this.

andreabolognani commented:
/test pull-kubevirt-e2e-k8s-1.22-sig-compute

andreabolognani commented:
The contents of the libguestfs appliance were apparently not the problem, though we'll still want to switch that over to CentOS Stream 9 too.

I'll keep digging :)

andreabolognani commented:
/test pull-kubevirt-build
/test pull-kubevirt-e2e-k8s-1.22-sig-compute

germag commented on hack/rpm-deps.sh:
```diff
@@ -86,7 +86,7 @@ launcherbase_base="
 "
```
I know that virtiofsd is not officially supported (yet), but the current launcher image includes virtiofsd (it's part of qemu-kvm-core). However, in CentOS 9, virtiofsd is packaged independently.

Since the C version is now deprecated in favor of the Rust version, could the virtiofsd-1.4.0 package be included? (for both aarch64 and x86_64).

andreabolognani replied:
@germag good catch, I'll take care of that :)

andreabolognani replied:
Done!

andreabolognani commented:
The Python problem was caused by `python` and `python3` being symlinks to `python3.6` - this was correct for CentOS Stream 8, but they need to point to `python3.9` for CentOS Stream 9.

We still need to figure out the bazel cache situation.

andreabolognani commented:
/retest

xpivarc commented:
@AlonaKaplan @EdDev Would you be able to help with network failures?

andreabolognani commented:
Bumped package versions a bit further.

/retest

vladikr commented:
`AC97 is not a valid device model name`
Looks like we need to find another device that we can safely passthrough

andreabolognani commented:
> First create a `user.bazelrc` file and tell it where it should cache the artifacts:
> 
> ```
> cat <<EOT >> user.bazelrc
> build --disk_cache=~/.localcache
> run --disk_cache=~/.localcache
> test --disk_cache=~/.localcache
> EOT
> ```
> 
> Then checkout current `main` and run `make && make build-functest && make test`, then switch to your branch and run `make && make build-functest && make test` again.

@rmohr I had to run `make` a few extra times after switching branches:

```
$ make
./hack/dockerized "hack/bazel-fmt.sh"
hack/dockerized "export BUILD_ARCH= && export DOCKER_TAG= && hack/bazel-fmt.sh && hack/bazel-build.sh"
hack/manifests.sh
Building manifests...
go version go1.17.8 linux/amd64
go version go1.17.8 linux/amd64
go version go1.17.8 linux/amd64



rsync: [receiver] stat "/.git/.index.OaF4cr" (in build) failed: No such file or directory (2)
rsync: [receiver] rename "/.git/.index.OaF4cr" (in build) -> ".git/index": No such file or directory (2)
WARNING: rpm/BUILD.bazel failed verification -- update discarded (will try again).
Bazel server is outdated, restarting ...
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1327) [sender=3.2.6]
make: *** [Makefile:16: bazel-build] Error 23
make: *** Waiting for unfinished jobs....
Bazel server is outdated, restarting ...
kubevirt-bazel-server
kubevirt-bazel-server
docker: Error response from daemon: Conflict. The container name "/kubevirt-bazel-server" is already in use by container "e4983d53b743a2cc6efdb29e725394d079d7034ea77644bdae88e6ab4ecd5531". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.
docker: Error response from daemon: Conflict. The container name "/kubevirt-bazel-server" is already in use by container "e4983d53b743a2cc6efdb29e725394d079d7034ea77644bdae88e6ab4ecd5531". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.
make: *** [Makefile:129: manifests] Error 125
make: *** [Makefile:187: format] Error 125
$ make
./hack/dockerized "hack/bazel-fmt.sh"
hack/dockerized "export BUILD_ARCH= && export DOCKER_TAG= && hack/bazel-fmt.sh && hack/bazel-build.sh"
hack/manifests.sh
Building manifests...
go version go1.17.8 linux/amd64
go version go1.17.8 linux/amd64
go version go1.17.8 linux/amd64



docker: Error response from daemon: Conflict. The container name "/kubevirt-bazel-server" is already in use by container "8ca512c3bffc73b78ef16a51a6ac22d93b30cca00a11b7d7f91d6fcd86b8d4d2". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.
make: *** [Makefile:187: format] Error 125
make: *** Waiting for unfinished jobs....
docker: Error response from daemon: Conflict. The container name "/kubevirt-bazel-server" is already in use by container "8ca512c3bffc73b78ef16a51a6ac22d93b30cca00a11b7d7f91d6fcd86b8d4d2". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.
make: *** [Makefile:129: manifests] Error 125
8ca512c3bffc73b78ef16a51a6ac22d93b30cca00a11b7d7f91d6fcd86b8d4d2
the input device is not a TTY
make: *** [Makefile:16: bazel-build] Error 1
```

After that, it seemed to have worked...

```
$ make
./hack/dockerized "hack/bazel-fmt.sh"
hack/dockerized "export BUILD_ARCH= && export DOCKER_TAG= && hack/bazel-fmt.sh && hack/bazel-build.sh"
hack/manifests.sh
Building manifests...
go version go1.17.8 linux/amd64
go version go1.17.8 linux/amd64
go version go1.17.8 linux/amd64



go version go1.17.8 linux/amd64
the input device is not a TTY
                             Regenerating sandbox
the input device is not a TTY
                             make: *** [Makefile:129: manifests] Error 1
                                                                        make: *** Waiting for unfinished jobs....
                                                                                                                 make: *** [Makefile:16: bazel-build] Error 1
                                                                                                                                                             INFO: Invocation ID: 57043940-fadf-4e8c-b05b-06c08f330ba8
INFO: Analyzed target //:gazelle (67 packages loaded, 7928 targets configured).
INFO: Found 1 target...
Target //:gazelle up-to-date:
  bazel-bin/gazelle-runner.bash
  bazel-bin/gazelle
INFO: Elapsed time: 2.084s, Critical Path: 0.34s
INFO: 5 processes: 4 remote cache hit, 1 internal.
INFO: Build completed successfully, 5 total actions
[...]
```

... but running `make test` still produces the same error we're seeing in CI.

```
$ make test
hack/dockerized "hack/bazel-fmt.sh && CI= ARTIFACTS= hack/bazel-test.sh"
go version go1.17.8 linux/amd64

go version go1.17.8 linux/amd64
Sandbox is up to date
INFO: Invocation ID: 3b612cd1-1e3d-44b6-b4b9-22116dbc018c
INFO: Analyzed target //vendor/mvdan.cc/sh/v3/cmd/shfmt:shfmt (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //vendor/mvdan.cc/sh/v3/cmd/shfmt:shfmt up-to-date:
  bazel-bin/vendor/mvdan.cc/sh/v3/cmd/shfmt/shfmt_/shfmt
INFO: Elapsed time: 0.406s, Critical Path: 0.30s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
INFO: Invocation ID: 14aee953-d5f1-4672-b3c5-e01472cb40c1
INFO: Analyzed target //:gazelle (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //:gazelle up-to-date:
  bazel-bin/gazelle-runner.bash
  bazel-bin/gazelle
INFO: Elapsed time: 0.225s, Critical Path: 0.06s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
INFO: Invocation ID: f921d0d0-13fe-404c-b73b-b651e6aab5ed
INFO: Analyzed target //:goimports (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //:goimports up-to-date:
  bazel-bin/goimports.bash
INFO: Elapsed time: 0.158s, Critical Path: 0.06s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
INFO: Invocation ID: 75d122b3-689d-4d58-8874-b090ea83fedd
INFO: Analyzed target //:buildifier (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //:buildifier up-to-date:
  bazel-bin/buildifier.bash
INFO: Elapsed time: 0.156s, Critical Path: 0.06s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
Sandbox is up to date
INFO: Invocation ID: 4b9373b2-3bef-4449-9f09-2b68ef1921af
INFO: Build options --action_env and --features have changed, discarding analysis cache.
DEBUG: /root/.cache/bazel/_bazel_root/6f347497f91c9a385dcd9294645b76e0/external/io_bazel_rules_go/go/private/context.bzl:505:14: WARNING: --features=race is no longer supported. Use --@io_bazel_rules_go//go/config:race instead.
INFO: Analyzed 469 targets (1388 packages loaded, 22234 targets configured).
INFO: Found 345 targets and 124 test targets...
ERROR: /root/go/src/kubevirt.io/kubevirt/cmd/container-disk-v2alpha/BUILD.bazel:3:10: Compiling cmd/container-disk-v2alpha/main.c failed: undeclared inclusion(s) in rule '//cmd/container-disk-v2alpha:container-disk':
this rule is missing dependency declarations for the following files included by 'cmd/container-disk-v2alpha/main.c':
  '/usr/lib/gcc/x86_64-redhat-linux/8/include/stdarg.h'
  '/usr/lib/gcc/x86_64-redhat-linux/8/include/stddef.h'
  '/usr/lib/gcc/x86_64-redhat-linux/8/include/stdbool.h'
INFO: Elapsed time: 4.773s, Critical Path: 1.99s
INFO: 1461 processes: 1224 remote cache hit, 223 internal, 14 processwrapper-sandbox.
[...]
```

Can I try anything else?

andreabolognani commented:
> `AC97 is not a valid device model name`
> Looks like we need to find another device that we can safely passthrough

Yeah, this seems to have been purposefully disabled as part of [RHBZ#1995819](https://bugzilla.redhat.com/show_bug.cgi?id=1995819).

It *sounds* (get it? ;) like `ich9` could be a possible replacement, but I don't know anything about the specific KubeVirt test case this change is breaking, or indeed audio support in QEMU O:-)

andreabolognani commented:
> Ok, great! I think I get it now. It looks like our custom GCC toolchains seem to look the same from the bazel perspective. We probably have to change an attribute on it (maybe the name), so that bazel realized that artifacts crated by the toolchain are not identical anymore, since the header locations changed. On meetings for the rest of the day, but can spend a little bit of time on it tomorrow morning. I think we will have resolved that soon. Thanks for your experiments!

Awesome! If you can get that sorted out, we'll be one massive step closer to getting this merged :)

rmohr commented:
> > Ok, great! I think I get it now. It looks like our custom GCC toolchains seem to look the same from the bazel perspective. We probably have to change an attribute on it (maybe the name), so that bazel realized that artifacts crated by the toolchain are not identical anymore, since the header locations changed. On meetings for the rest of the day, but can spend a little bit of time on it tomorrow morning. I think we will have resolved that soon. Thanks for your experiments!
> 
> Awesome! If you can get that sorted out, we'll be one massive step closer to getting this merged :)

So, quickly asked on bazel slack, and our situation makes a lot of sense: these system header files are not considered for the bazel action cache because it is intransparent for bazel if any of the headers in these directories are ever added as a dependency. In order to invalidate on such changes all caches, we can just add a new flag to the toolchain and change its value whenever we update our toolchain configuration.

A more future-proof solution would be to extra specify our custom sysroot as dependency but I had some path configuration issues in the past.

So we can add in https://github.com/kubevirt/kubevirt/blob/main/bazel/toolchain/x86_64-none-linux-gnu/cc_toolchain_config.bzl#L75 a no-op flag where we can bump the value together with such a bump. For example:

```
                            "-D__TOOLCHAIN_SYSROOT__=\"centos9\"",
```

andreabolognani commented:
> So we can add in https://github.com/kubevirt/kubevirt/blob/main/bazel/toolchain/x86_64-none-linux-gnu/cc_toolchain_config.bzl#L75 a no-op flag where we can bump the value together with such a bump. For example:
> 
> ```
>                             "-D__TOOLCHAIN_SYSROOT__=\"centos9\"",
> ```

Implemented this suggestion and dropped the commit where I disabled the cache. Let's see if it does the trick.

/retest

Assuming that works, I'll start looking into fixing some of the genuine failures that have been reported by functional tests.

rmohr commented:
> > So we can add in https://github.com/kubevirt/kubevirt/blob/main/bazel/toolchain/x86_64-none-linux-gnu/cc_toolchain_config.bzl#L75 a no-op flag where we can bump the value together with such a bump. For example:
> > ```
> >                             "-D__TOOLCHAIN_SYSROOT__=\"centos9\"",
> > ```
> 
> Implemented this suggestion and dropped the commit where I disabled the cache. Let's see if it does the trick.
> 
> /retest
> 
> Assuming that works, I'll start looking into fixing some of the genuine failures that have been reported by functional tests.

Results look promising :tada:

rmohr commented on third_party/toolchains/toolchains.bzl:
```diff
@@ -8,9 +8,8 @@ def toolchains():
-            sha256 = "fe7f72330216612de44891ebe5e228eed7c0c051ac090c395b2b33115c6f5408",
-            strip_prefix = "gcc-arm-10.2-2020.11-x86_64-aarch64-none-linux-gnu",
+            sha256 = "50cdef6c5baddaa00f60502cc8b59cc11065306ae575ad2f51e412a9b2a90364",
+            strip_prefix = "arm-gnu-toolchain-11.3.rel1-x86_64-aarch64-none-linux-gnu",
```
FYI, because we pull in the arm toolchain as an external artifact, bazel sees the change and can do the right thing for arm.

andreabolognani replied:
So do you think I should rip the aarch64 part out as it's not strictly needed, or is it better to have it anyway for consistency? I'm fine either way.

vladikr commented:
> > `AC97 is not a valid device model name`
> > Looks like we need to find another device that we can safely passthrough
> 
> Yeah, this seems to have been purposefully disabled as part of [RHBZ#1995819](https://bugzilla.redhat.com/show_bug.cgi?id=1995819).
> 
> It _sounds_ (get it? ;) like `ich9` could be a possible replacement, but I don't know anything about the specific KubeVirt test case this change is breaking, or indeed audio support in QEMU O:-)

We used this virtual sound card to test PCI passthrough. It doesn't really to work as a sound card, but it was used as dummy PCI device.

andreabolognani commented:
> We used this virtual sound card to test PCI passthrough. It doesn't really to work as a sound card, but it was used as a dummy PCI device.

@vladikr I see that the test cases in question come in several flavors: default (ich9), ich9 and ac97. Do you think it would be okay to drop the ac97 test case, as the functionality is being covered by the ich9 test case anyway?

We are, however, effectively making the ac97 device unavailable with this update. Should the model be rejected at the KubeVirt level? Is there a process in place to deprecate/remove specific devices?

vladikr commented:
> > > We used this virtual sound card to test PCI passthrough. It doesn't really to work as a sound card, but it was used as a dummy PCI device.
> > 
> > 
> > @vladikr I see that the test cases in question come in several flavors: default (ich9), ich9 and ac97. Do you think it would be okay to drop the ac97 test case, as the functionality is being covered by the ich9 test case anyway?
> 
> Yes. We have 2 different ac97 tests, and I believe @vladikr was referring to the other one (https://github.com/kubevirt/kubevirt/blob/main/tests/vmi_hostdev_test.go#L97), which doesn't seem to be a problem here. (though it probably will be a problem when we update kubevirtci to Centos 9, but we'll just change it in favor of a supported device then). AC97 is really old, and it doesn't surprise me that libvirt/qemu dropped support for it. I vote for removing that test.

@jean-edouard yeah, you're absolutely right, I was referring to another test... thanks!

> 
> > We are, however, effectively making the ac97 device unavailable with this update. Should the model be rejected at the KubeVirt level? Is there a process in place to deprecate/remove specific devices?
> 
> That's a good point. I'd suggest filing a github issue for documenting/handling the fact that AC97 is now unsupported.

@andreabolognani we have a validation webhook[1], we can remove ac97 from there, but AFAIK we don't have a device depreciation process.
Yeah, We should let users know that it's not supported with the centos9 update, either by updating the doc or via the mailing list.


[1] https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go#L820

andreabolognani commented:
> Yes. We have 2 different ac97 tests, and I believe @vladikr was referring to the other one (https://github.com/kubevirt/kubevirt/blob/main/tests/vmi_hostdev_test.go#L97), which doesn't seem to be a problem here. (though it probably will be a problem when we update kubevirtci to Centos 9, but we'll just change it in favor of a supported device then).

Yeah, in particular we'll have to change [the `vm.sh` script](https://github.com/kubevirt/kubevirtci/blob/bcdea3ee1b0fa99aecafa1256d1e231d532f1dab/cluster-provision/centos8/scripts/vm.sh#L153) to use something else instead of ac97. ich9 feels like the obvious candidate. We don't need to do anything right now though, since the functional test uses PCI device assignment and so it's agnostic to the model of the underlying device.

> we have a validation webhook[1], we can remove ac97 from there, but AFAIK we don't have a device depreciation process.
> Yeah, We should let users know that it's not supported with the centos9 update, either by updating the doc or via the mailing list.

The tricky question is: should ac97 support be completely removed from KubeVirt? Update the validation webhook so that it rejects it, drop all other logic related to it as well as mentions of it in the schema and documentation?

On one hand, that would result in the best experience for people installing official releases. On the other hand, downstreams other than Red Hat, who produce KubeVirt images based on their own distro and their own QEMU builds, in which ac97 support might still be present, would be losing access to the device for no obvious reason...

I don't know what the correct answer is here, and I feel that it's not me up to make this call anyway. Hopefully the maintainers can weigh in :)

In the meantime I'll drop the test, which we all seem to agree is not buying us much anyway.

rmohr commented:
> The tricky question is: should ac97 support be completely removed from KubeVirt? Update the validation webhook so that it rejects it, drop all other logic related to it as well as mentions of it in the schema and documentation?

What is the `ac97` replacement? Let's at least check with the original authors of this patch if they can live with a replacement? It is an API breakage, which we normally don't do. :/

jean-edouard commented:
The failure on the storage lanes seems to be caused by the following SELinux denial:
```
type=AVC msg=audit(1665569998.736:3498): avc:  denied  { ioctl } for  pid=83966 comm="iptables" path="/sys/fs/cgroup" dev="tmpfs" ino=85 scontext=system_u:system_r:iptables_t:s0 tcontext=system_u:object_r:cgroup_t:s0 tclass=dir permissive=0
```
That failure used to happen on Fedora 24 Core OS, as reported by this bug:
https://bugzilla.redhat.com/show_bug.cgi?id=2008097
And fixed by this PR:
https://github.com/fedora-selinux/selinux-policy/pull/1083

I guess we need a similar PR against the Centos 9 SELinux policy?...

rmohr commented:
> Edit: actually, the policy in use is the kernel policy, which is still Centos 8. I suggest we quarantine that test and file an issue to un-quarantine it after updating CI to Centos 9.

Wouldn't that mean that we implicitly say that KubeVirt will not work on Centos 8 nodes anymore?

zpytela commented:
> I guess we need a similar PR against the Centos 9 SELinux policy?...
> 
> Edit: actually, the policy in use is the kernel policy, which is still Centos 8. I suggest we quarantine that test and file an issue to un-quarantine it after updating CI to Centos 9.

Bugzillas needs to be open (cloned from the fedora one) for RHEL/Centos 8 and 9.

victortoso commented:
> Victor, what was the rationale behind implementing support for ac97 in addition to the default, more modern ich9? Did you feel that many people might need to choose the older device for some reason?

For several years, we had windows guest bugs with ich* audio devices that did not happen with ac97 plus proprietary drivers ([example](https://bugzilla.redhat.com/show_bug.cgi?id=1203507)). That's the main reason I opted to include ac97 as well as an alternative.

jean-edouard commented:
> > I guess we need a similar PR against the Centos 9 SELinux policy?...
> > Edit: actually, the policy in use is the kernel policy, which is still Centos 8. I suggest we quarantine that test and file an issue to un-quarantine it after updating CI to Centos 9.
> 
> Bugzillas needs to be open (cloned from the fedora one) for RHEL/Centos 8 and 9.

Thanks a lot for chiming in @zpytela ! I'll open a BZ for Centos 8, and I'll check if Centos 9 also needs it.

andreabolognani commented:
> For several years, we had windows guest bugs with ich* audio devices that did not happen with ac97 plus proprietary drivers ([example](https://bugzilla.redhat.com/show_bug.cgi?id=1203507)). That's the main reason I opted to include ac97 as well as an alternative.

@victortoso that specific bug has been resolved three years ago. I know it was just an example, but surely the situation has only improved further since then?

Red Hat seems to be confident enough in the current state of ich9, including on Windows (which we fully support as a guest), to drop ac97 support from QEMU. So perhaps we can assume that most of the issues have been resolved by now, and any remaining ones are going to be addressed promptly in CentOS Stream 9?

andreabolognani commented:
Disabled the ac97 test and reintroduced a libguestfs appliance based on CentOS Stream 9.

/retest

andreabolognani commented:
/test pull-kubevirt-unit-test

andreabolognani commented:
Dropped the CentOS Stream 9-based libguestfs appliance: discussion on the exact approach to take is still ongoing, and I don't currently have a way to host the alternative image semi-permanently.

Added an alternative passt build that is patched for compatibility with CentOS Stream 9's libvirt version.

Pointed to a patched version of bazeldnf that is able to pick recent passt builds.

/retest

jean-edouard commented:
@andreabolognani if you'd like to test if there's anything else blocking the network lanes, you can try adding (in a DNM commit) the following line at the very end of `cmd/virt-handler/virt_launcher.cil`:
`(allow iptables_t cgroup_t (dir (ioctl)))`
Probably best to catch any other blocker as early as possible, as that SELinux issue could take a while to fix. Btw:
Centos 8 Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2134820
Centos 9 Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2134829

andreabolognani commented:
> if you'd like to test if there's anything else blocking the network lanes, you can try adding (in a DNM commit) the following line at the very end of `cmd/virt-handler/virt_launcher.cil`:
> `(allow iptables_t cgroup_t (dir (ioctl)))`

@jean-edouard excellent tip! I was about to start looking into the networking issues, but since it's far from my area of expertise I didn't quite know where to start. Now I have a lead :)

Just to make sure we're on the same page, would merging such a workaround until the upstream SELinux policy has been updated not be acceptable? Looking at the file, it seems like that's kind of its intended purpose...

rmohr commented:
> Just to make sure we're on the same page, would merging such a workaround until the upstream SELinux policy has been updated not be acceptable? Looking at the file, it seems like that's kind of its intended purpose...

Yes, that is exactly its purpose :)

zpytela commented:
> > Just to make sure we're on the same page, would merging such a workaround until the upstream SELinux policy has been updated not be acceptable? Looking at the file, it seems like that's kind of its intended purpose...
> 
> Yes, that is exactly its purpose :)

Correct, and the selinux-policy fix will be a superset of this single allow rule. F36 and newer have already been updated.

AlonaKaplan commented:
> @AlonaKaplan @EdDev Would you be able to help with network failures?

Sorry, I was on a PTO. Do you still need help?

jean-edouard commented:
> > Just to make sure we're on the same page, would merging such a workaround until the upstream SELinux policy has been updated not be acceptable? Looking at the file, it seems like that's kind of its intended purpose...
> 
> Yes, that is exactly its purpose :)

Ahem yes... But all we do (for now) is create a new type that's used only by us, instead of modifying container_t for everybody which would be much easier but extremely invasive.
Pushing this additional rule would change the iptables policy for the whole system, which doesn't feel right.

That being said, this Centos 9 update is very important and many users are eagerly waiting for it.
Also, the rule in question has already been approved for Fedora and will most likely be approved in Centos.

Let's add it and see what happens between now and when this PR will be ready to merge.
If we're lucky we won't need it anymore by then anyway!

rmohr commented:
> > > Just to make sure we're on the same page, would merging such a workaround until the upstream SELinux policy has been updated not be acceptable? Looking at the file, it seems like that's kind of its intended purpose...
> > 
> > 
> > Yes, that is exactly its purpose :)
> 
> Ahem yes... But all we do (for now) is create a new type that's used only by us, instead of modifying container_t for everybody which would be much easier but extremely invasive. Pushing this additional rule would change the iptables policy for the whole system, which doesn't feel right.
> 
> That being said, this Centos 9 update is very important and many users are eagerly waiting for it. Also, the rule in question has already been approved for Fedora and will most likely be approved in Centos.
> 
> Let's add it and see what happens between now and when this PR will be ready to merge. If we're lucky we won't need it anymore by then anyway!

Btw. Why do we need this at all? Does not look like we needed it before with the same amount of functionality ...

andreabolognani commented:
Picked up bazeldnf v5.4.0 and tried adding the SELinux workaround. Let's see whether that actually helps.

There are at least a couple of cases where I'm pretty sure the error is unrelated to SELinux and is connected to QEMU command line generation instead. I'm going to look into those next.

/retest

jean-edouard commented:
> > > > Just to make sure we're on the same page, would merging such a workaround until the upstream SELinux policy has been updated not be acceptable? Looking at the file, it seems like that's kind of its intended purpose...
> > > 
> > > 
> > > Yes, that is exactly its purpose :)
> > 
> > 
> > Ahem yes... But all we do (for now) is create a new type that's used only by us, instead of modifying container_t for everybody which would be much easier but extremely invasive. Pushing this additional rule would change the iptables policy for the whole system, which doesn't feel right.
> > That being said, this Centos 9 update is very important and many users are eagerly waiting for it. Also, the rule in question has already been approved for Fedora and will most likely be approved in Centos.
> > Let's add it and see what happens between now and when this PR will be ready to merge. If we're lucky we won't need it anymore by then anyway!
> 
> Btw. Why do we need this at all? Does not look like we needed it before with the same amount of functionality ...

Looks like the SELinux denial is not what's the causing the network tests to fail. Maybe the denial was there all along.

andreabolognani commented:
> Looks like the SELinux denial is not what's the causing the network tests to fail. Maybe the denial was there all along.

Dropped the SELinux change based on these results.

Fixed slirp command line generation: this doesn't seem to fix the corresponding test cases completely, but at least things go a little bit further. I'll keep investigating.

/retest pull-kubevirt-e2e-k8s-1.22-sig-network

andreabolognani commented:
/test pull-kubevirt-e2e-k8s-1.22-sig-network

andreabolognani commented:
The slirp tests should pass now. Whether or not the way I've achieved this is the most reasonable one is up for debate O:-)

/test pull-kubevirt-e2e-k8s-1.22-sig-network

andreabolognani commented:
Many of the network tests were failing because the connectivity check is performed using `nc`, which suffered from the lack of crypto-policies configuration.

I have added the necessary symlinks, so most things should work now. Fingers crossed!

/retest

andreabolognani commented:
Fixed a couple of style issues reported by the linter and updated passt to the very latest release after introducing another workaround in the custom build. Also refreshed the CentOS Stream 9-based builder image, which was getting somewhat stale.

There's still at least one remaining test failure that looks genuine. I'll look into it next week.

/retest

andreabolognani commented:
Switched to the latest official build of passt, which includes the fixes we need for CentOS Stream 9, and bumped the version number for a few other components.

/retest

andreabolognani commented:
This should address the remaining failures on the network side.

/retest

There's still a problem with mdev handling:

```
[Serial][sig-compute]MediatedDevices with mediated devices  configuration Should successfully passthrough a mediated device

Failure tests/mdev_configuration_allocation_test.go:166
Timed out after 120.001s.
wait for the kubelet to stop promoting unconfigured devices
Expected
    <int64>: 16
to be zero-valued
tests/mdev_configuration_allocation_test.go:167
```

I believe what happens is that the mdevs are created correctly, but do not get disposed of properly when no longer needed.

Unfortunately I think reproducing the failure locally would require having mdev-capable hardware handy, and I don't currently have convenient access to that. @vladikr @rmohr can you think of someone who would be able to look into this?

andreabolognani commented:
More errors with istio + passt are being reported now. They were skipped in previous runs. Is it possible that failures early in the lane would cause tests that are later in the lane to be skipped?

Either way, I'll try to reproduce those locally as well.

jean-edouard commented:
> More errors with istio + passt are being reported now. They were skipped in previous runs. Is it possible that failures early in the lane would cause tests that are later in the lane to be skipped?
> 
> Either way, I'll try to reproduce those locally as well.

We run all the tests that can run in parallel first, then the serial ones, *unless* the parallel run failed.
That's usually how you see new failures after fixing issues.

andreabolognani commented:
Bumped passt to the very latest available version. This has fixed all of the istio + passt test cases for me.

/retest

andreabolognani commented:
Stopped skipping some tests on passt, since they should work when using the latest version.

There are a bunch of other failures being reported, but they mostly look like flakes. In a lot of cases the git tree is being reported as dirty, but I'm positive I've committed every change and double-checking just confirmed this.

A proper build of the libguestfs appliance has been created as part of https://github.com/kubevirt/libguestfs-appliance/pull/13, so I was able to drop one of the DONOTMERGEs.

@rmohr @vladikr @xpivarc can one of you please build and publish a proper version of the builder container? Then I'll be able to drop the other DONOTMERGE as well, and finally mark the PR as ready for review. Thanks!

/retest

andreabolognani commented:
> There are a bunch of other failures being reported, but they mostly look like flakes. In a lot of cases the git tree is being reported as dirty, but I'm positive I've committed every change and double-checking just confirmed this.

I take this back: I had committed changes out of order, so CI was right to complain. It should all be okay now.

/retest

jean-edouard commented:
> @rmohr @vladikr @xpivarc can one of you please build and publish a proper version of the builder container? Then I'll be able to drop the other DONOTMERGE as well, and finally mark the PR as ready for review. Thanks!

I can do that in a couple hours

jean-edouard commented:
> > @rmohr @vladikr @xpivarc can one of you please build and publish a proper version of the builder container? Then I'll be able to drop the other DONOTMERGE as well, and finally mark the PR as ready for review. Thanks!
> 
> I can do that in a couple hours

Here you go! `quay.io/kubevirt/builder:2210260829-e03914a71`

andreabolognani commented:
> Here you go! `quay.io/kubevirt/builder:2210260829-e03914a71`

Thanks a lot! Let's see how CI likes it :)

/retest

AlonaKaplan commented on rpm/repo.yaml:
```diff
@@ -1,33 +1,33 @@
 - arch: x86_64
-  baseurl: https://download.copr.fedorainfracloud.org/results/sbrivio/passt/centos-stream-8-x86_64
-  name: centos/stream8-passt-x86_64
+  baseurl: https://download.copr.fedorainfracloud.org/results/sbrivio/passt/centos-stream-9-x86_64
```
The rpms in this repo are removed once a new version is uploaded.
Please use the mirror repo  - https://github.com/kubevirt/kubevirt/pull/8403#issuecomment-1290394953

andreabolognani replied:
@AlonaKaplan the RPMs from CentOS Stream itself are also going to be cycled out as new packages are introduced. That's exactly why we mirror everything. So I'm not sure we need to treat passt any different.

Moreover, I'm concerned that, if we start pointing to the passt.top mirror, @sbrivio-rh will end up having to foot a quite substantial bandwidth bill. Let's hear what he thinks about that.

andreabolognani replied:
Also the COPR situation is just temporary: passt is already in Fedora, and it shouldn't be long before it is added to CentOS Stream 9 too, at which point we will be able to pick it up the same as any other package.

sbrivio-rh replied:
> Moreover, I'm concerned that, if we start pointing to the passt.top mirror, @sbrivio-rh will end up having to foot a quite substantial bandwidth bill. Let's hear what he thinks about that.

Don't worry about that, it's not in "the cloud", I have cheap, unlimited bandwidth.

> passt is already in Fedora, and it shouldn't be long before it is added to CentOS Stream 9 too, at which point we will be able to pick it up the same as any other package.

...also true. My understanding is that we should be a few days away. The process has been going on for quite a while now.

sbrivio-rh replied:
While we're here, note that there's a Copr issue which also affects my mirror:
* latest available build is 04992985, see e.g. https://download.copr.fedorainfracloud.org/results/sbrivio/passt/srpm-builds/04992985
* ...but that directory is unreachable from the index, see: https://download.copr.fedorainfracloud.org/results/sbrivio/passt/srpm-builds/

so my mirror can't pick it up either. That usually gets fixed in a few hours, but sometimes in about one day.

I'm trying to find a workaround. I swear I've seen an issue filed at https://pagure.io/fedora-infrastructure/issues, but I can't find it anymore. I might file a new ticket.

sbrivio-rh replied:
Filed https://pagure.io/fedora-infrastructure/issue/10960.

sbrivio-rh replied:
> While we're here, note that there's a Copr issue which also affects my mirror:
> 
>     * latest available build is 04992985, see e.g. https://download.copr.fedorainfracloud.org/results/sbrivio/passt/srpm-builds/04992985
> 
>     * ...but that directory is unreachable from the index, see: https://download.copr.fedorainfracloud.org/results/sbrivio/passt/srpm-builds/
> 
> 
> so my mirror can't pick it up either. That usually gets fixed in a few hours, but sometimes in about one day.

Solved now, I wasn't aware of CDN caching:
https://docs.pagure.org/copr.copr/user_documentation.html#build-succeeded-but-i-don-t-see-the-built-results

The mirror now uses non-cached sources.

andreabolognani replied:
Switched to the mirror in the latest version.

andreabolognani commented:
Addressed all review comments.

Bumped the libvirt Go bindings too, even though the new version doesn't introduce any significant changes.

/retest

andreabolognani commented:
I'm not sure whether this is a flake or a consequence of bumping passt from `0^20221024.gc11277b` to `0^20221026.gf212044`. If this keeps failing, I'll downgrade the package and we can look into the issue separately, without holding up this PR further.

/test pull-kubevirt-e2e-kind-1.22-sriov

sbrivio-rh commented:
> I'm not sure whether this is a flake or a consequence of bumping passt from `0^20221024.gc11277b` to `0^20221026.gf212044`.

I had a look, but it's not clear to me what the network connections are. Even though in the SRIOV test, just by looking at comments and test descriptions, I would reasonably assume it's a direct SRIOV connection. In the other ones I can't understand if traffic is going through passt.

Changes between those two versions are very limited and very isolated from any TCP path.

sbrivio-rh commented:
Hm, okay, so the `-sriov` thing succeeded now...?

/retest, quick!

sbrivio-rh commented:
/retest

sbrivio-rh commented:
Hm, okay, so the `-sriov` thing succeeded now...?

/retest, quick!

andreabolognani commented:
@sbrivio-rh yeah it looks like the test was just flaky after all :)

@rmorh @vladikr @xpivarc  @jean-edouard anyone willing to LGTM this? Looks like there's nothing stopping it from being merged at this point.

andreabolognani commented:
Thanks @xpivarc!

One thing that I just noticed. The bot is reporting a few jobs as having failed, specifically

* `pull-kubevirt-e2e-k8s-1.23-sig-network-nonroot`
* `pull-kubevirt-e2e-k8s-1.23-sig-compute-nonroot`
* `pull-kubevirt-e2e-k8s-1.23-operator-nonroot`
* `pull-kubevirt-e2e-k8s-1.23-sig-storage-nonroot`

but these jobs no longer exist after https://github.com/kubevirt/project-infra/commit/e832710d2e3519985102b2b54ee946d6f7ee0a35.

I think prow somehow gets things mixed up when jobs that used to exist when the PR was created are removed. I wonder if something could be done about that? The current situation can be confusing to developers.

Then again, maybe it doesn't matter that much, because PRs that are as long-lived as this one are hopefully fairly uncommon :)

andreabolognani commented:
Looking at the PR history, the `pull-kubevirt-e2e-kind-1.23-vgpu-nonroot` job seems pretty flaky - it has both passed and failed for previous versions. Let's try again.

/test pull-kubevirt-e2e-kind-1.23-vgpu-nonroot

vladikr commented:
Thank you @andreabolognani !
